MsCorePkg/MacAddressEmulationDxe/MacAddressEmulationDxe.c

```diff
@@ -0,0 +1,278 @@
+/** @file
+
+  DXE Driver for handling MAC Address Emulation.
+
+  Copyright (C) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+#include "MacAddressEmulationDxe.h"
+
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/DebugLib.h>
+#include <Library/UefiLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/NetLib.h>
+
+#include <Library/MacAddressEmulationPlatformLib.h>
+
+/**
+  Performs sanity checks to ensure an snp can support mac emulation, and ensures that multiple interfaces are not programmed.
+
+  @param[in] SnpHandle - A handle to an Snp
+  @param[in] Snp - An snp protocol associated with the handle provided
+  @param[in] SnpContext - The snp context created by this driver's entry point
+  @retval TRUE - the SNP supports mac emulation and can be programmed with the emulated address
+  @retval FALSE - the SNP should not be programmed with the emulated address
+
+  @remark  typically called at TPL_NOTIFY
+**/
+BOOLEAN
+SnpSupportsMacEmuCheck (
+  IN CONST EFI_HANDLE                        SnpHandle,
+  IN CONST EFI_SIMPLE_NETWORK_PROTOCOL       *Snp,
+  IN CONST MAC_EMULATION_SNP_NOTIFY_CONTEXT  *Context
+  )
+{
+  BOOLEAN  IsMatch = FALSE;
+
+  if ((SnpHandle == NULL) || (Snp == NULL) || (Context == NULL)) {
+    return FALSE;
+  }
+
+  if (Snp->Mode->State != EfiSimpleNetworkInitialized) {
+    DEBUG ((DEBUG_WARN, "[%a]: SNP handle in unexpected state %d, cannot update MAC.\n", __FUNCTION__, Snp->Mode->State));
+    return FALSE
+  }
+
+  if (Snp->Mode->IfType != NET_IFTYPE_ETHERNET) {
+    DEBUG ((DEBUG_WARN, "[%a]: SNP interface type is not Ethernet.\n", __FUNCTION__));
+    return FALSE;
+  }
+
+  if (!Snp->Mode->MacAddressChangeable) {
+    DEBUG ((DEBUG_WARN, "[%a]: SNP interface does not support MAC address programming", __FUNCTION__));
+    return FALSE;
+  }
+
+  if (!PlatformMacEmulationSnpCheck (SnpHandle)) {
+    DEBUG ((DEBUG_WARN, "[%a]: Platform library reports not to support this SNP", __FUNCTION__));
+    return FALSE;
+  }
+
+  if (IsMatch && (Context->Assigned == TRUE)) {

```

Comment:
Looks like you can drop `IsMatch` entirely now.

Reply:
Yeah forgot to delete that. Was letting the CI do the unit tests for me (which would have caught this error) 😁